test/ParseableInterface/ModuleCache/prebuilt-module-cache-forwarding.swift

```diff
@@ -0,0 +1,24 @@
+// RUN: %empty-directory(%t)
+// RUN: %empty-directory(%t/MCP)
+// RUN: %empty-directory(%t/prebuilt-cache)
+
+// First, prebuild a module and put it in the prebuilt cache.
+// RUN: sed -e 's/FromInterface/FromPrebuilt/g' %S/Inputs/prebuilt-module-cache/Lib.swiftinterface > %t/Lib.swiftinterface
+// RUN: %target-swift-frontend -parse-stdlib -module-cache-path %t/MCP -emit-module-path %t/prebuilt-cache/Lib.swiftmodule %t/Lib.swiftinterface -module-name Lib
+
+// Next, use the module and check if the forwarding module is in place.
+// RUN: not %target-swift-frontend -typecheck -enable-parseable-module-interface -parse-stdlib -module-cache-path %t/MCP -sdk %S/Inputs -I %S/Inputs/prebuilt-module-cache/ -prebuilt-module-cache-path %t/prebuilt-cache %s 2>&1 | %FileCheck -check-prefix=FROM-PREBUILT %s
+
+// A forwarding module is going to be a YAML file which will start with "---"
+// RUN: %{python} %S/Inputs/is-yaml-file.py %t/MCP/Lib-*.swiftmodule || not echo "could not find forwarding module in module cache '%t/MCP/'"
+
+// Now invalidate a dependency of the prebuilt module, and make sure the forwarding file is replaced with a real module.
+// RUN: touch %t/Lib.swiftinterface
+// RUN: not %target-swift-frontend -typecheck -enable-parseable-module-interface -parse-stdlib -module-cache-path %t/MCP -sdk %S/Inputs -I %S/Inputs/prebuilt-module-cache/ -prebuilt-module-cache-path %t/prebuilt-cache %s 2>&1 | %FileCheck -check-prefix=FROM-INTERFACE %s
```

Comment:
Also test what happens if you invalidate a dependency without the forwarding module present, and what happens if you have a forwarding module but the underlying module goes away for whatever reason.